### ISSUE TITLE
Write tests

### ISSUE THREAD

**Issue by rubik** [rubik/pyg#69]
We should write some tests for Pyg. I still don't know what test suite to use, but I would suggest `py.test` or `lettuce`. I never tried lettuce but it looks interesting!
We can leave this for Pyg 1.0.

**Comment by fdev31** [comment]
We may have a list of options 100% covered  for next version, then we will add less critic feature tests, don't you think ?

**Comment by rubik** [comment]
Yes, such as installing, removing, upgrading, requirement files, searching, and bundles. I think these are the most important features, aren't they?

**Comment by fdev31** [comment]
About the tests, I think we need to focus on the "test bed" first, making tests fast with "fake" packages in fake pypi server OR use a real http_proxy.
In all cases, we will probably need "100% controled" packages to test if they are running fine or not.
For instance, a test could do a pyg pack then expect some output calling run_pyg.py -h, things like that... but for other packages it may require some maintenance... what do you think ?

**Comment by rubik** [comment]
It is an interesting idea! :)
We could upload to PyPI fake packages like pyg-testegg, pyg-testsource, etc.
They will have the same interface, so that we can fully test them.

**Comment by fdev31** [comment]
I like this option, sounds better than embedding controled packages :)

**Comment by rubik** [comment]
Good! I'll open a new repository, maybe pyg-testpackages (is there a better name?) with a script to automate packages creation and upload.

**Comment by fdev31** [comment]
pyg-xp-tests ? pyg-xp-testpackages ?
I like the "xp", "wip", "fix", etc infix to sort the branches... don't you ?

**Comment by rubik** [comment]
I'll go for `pyg-xp-testpackages` (though I preferred `pyg-xp-tests` but it could be misleading).
